packages/react-components/react-shared-contexts/src/MaterialTypeContext/MaterialTypeContext.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+
+export type MaterialTypeContextValue = MaterialType;
+
+export const MaterialType = {
+  /**
+   * A solid surface that does not allow any light to pass through.
+   */
+  Opaque: 'opaque',
+  /**
+   * A surface that allows some light to pass through.
+   *
+   * > backdrop-filter: blur(80px);
+   */
+  SemiOpaque: 'semi-opaque',
+  /**
+   * A surface that allows light to pass through.
+   *
+   * > backdrop-filter: blur(45px);
+   */
+  Translucent: 'translucent',
+  /**
+   * A surface that allows light to pass through.
+   *
+   * > backdrop-filter: blur(30px);
+   */
+  SemiTransparent: 'semi-transparent',
+} as const;
+
+export type MaterialType = (typeof MaterialType)[keyof typeof MaterialType];
+
+namespace MaterialType {
```

Comment:
Why is the namespace necessary?

Reply:
it is not. I can have it removed, but if you need to refer to the type of a single element in a union created as a `const` then you'd have to access like `typeof MaterialType.UnionMember`, by using this name space exclusively for type you can do `MaterilType.UnionMember` work properly without having to use the `typeof` key and this does not produce anything on the JS bundle (as this namespace only includes types)